docs/src/developers/style.md

```diff
@@ -51,6 +51,220 @@ format("test")
 The following sections outline extra style guide points that are not fixed
 automatically by JuliaFormatter.
 
+### Abstract types and composition
+
+Specifying types for method arguments and struct fields is mostly optional in
+Julia. The benefit of this is that it enables functions and types from one
```

Comment:
The second sentence "The benefit of this ..." applies clearly for method arguments but is a bit confusing for struct fields.